js/pathfinding.js

```diff
@@ -61,8 +65,12 @@ function getCachedNodes(token) {
 
 	// Different-sized tokens snap to different points on the grid,
 	// so they might follow a different path to other tokens
-	cacheData.tokenWidth = token.width;
-	cacheData.tokenHeight = token.height;
+	if (canvas.grid.type === CONST.GRID_TYPES.SQUARE) {
+		cacheData.tokenWidth = token.width;
+		cacheData.tokenHeight = token.height;
+	} else if (canvas.grid.isHex && game.modules.get("hex-size-support")?.active) {
+		cacheData.tokenSize = token.document.data.flags["hex-size-support", "borderSize"];
```

Comment:
You also need to capture the value of the "alt snap flag", which indicates the orientation of the token when interfacing with `hex-size-support` (at least for tokens with an even size. It doesn't matter for unevenly sized tokens)

Reply:
Done this, as well as snap to vertex which could potentially make a difference. I hope nobody uses different settings for different tokens within the same map...